reliability-engineering/terraform/modules/concourse-team-iam-credentials/iam.tf

```diff
@@ -0,0 +1,57 @@
+resource "aws_iam_policy" "concourse_sts_rotation_lambda_execution" {
+  name = "${var.deployment}-concourse-sts-rotation-lambda-execution"
+
+  policy = <<-POLICY
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Action": [
+          "ssm:PutParameter"
+        ],
+        "Effect": "Allow",
+        "Resource": [
+          "arn:aws:ssm:eu-west-2:${data.aws_caller_identity.account.account_id}:parameter/${var.deployment}/concourse/pipelines/*/readonly_access_key_id",
+          "arn:aws:ssm:eu-west-2:${data.aws_caller_identity.account.account_id}:parameter/${var.deployment}/concourse/pipelines/*/readonly_secret_access_key",
+          "arn:aws:ssm:eu-west-2:${data.aws_caller_identity.account.account_id}:parameter/${var.deployment}/concourse/pipelines/*/readonly_session_token"
+        ]
+      }, {
+        "Effect": "Allow",
+        "Action": [
+          "kms:ListKeys",
+          "kms:ListAliases",
+          "kms:Describe*",
+          "kms:Decrypt",
+          "kms:Encrypt"
+        ],
+        "Resource": [
+          "${var.kms_key_arn}"
+        ]
+      }, {
+        "Effect": "Allow",
+        "Action": [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents",
+          "logs:DescribeLogStreams"
+        ],
+        "Resource": [
+          "arn:aws:logs:*:*:*"
+        ]
+      }, {
+        "Effect": "Allow",
+        "Action": [
+          "sts:AssumeRole"
+        ],
+        "Resource": "*"
```

Comment:
Could this be a refined, such as `arn:aws:iam::*:role/*-concourse-worker`?

Reply:
I tend to treat trust policies as the important place to control who can assume which role, and tend to be very broad with who is allowed to perform `sts:AssumeRole`; but I am not against making this change.

Reply:
oh wait I remember why I didn't do this: because all of the `*-concourse-worker` roles have `sts:AssumeRole` permissions for `"*"` in order to do their job (of doing cross-account deploys).  So restricting the AssumeRole here doesn't gain us anything, because you can always assume some worker role and then assume any other role to get around the restriction.